chef/cookbooks/neutron/recipes/l3_ha.rb

```diff
@@ -65,18 +65,26 @@
 end
 
 networking_plugin = node[:neutron][:networking_plugin]
+
 case networking_plugin
-when "openvswitch", "cisco"
-  neutron_agent = node[:neutron][:platform][:ovs_agent_name]
-when "linuxbridge"
-  neutron_agent = node[:neutron][:platform][:lb_agent_name]
+when 'ml2'
+  ml2_mech_drivers = node[:neutron][:ml2_mechanism_drivers]
+  case
+  when ml2_mech_drivers.include?("openvswitch")
+    neutron_agent = node[:neutron][:platform][:ovs_agent_name]
+    neutron_agent_ra = node[:neutron][:ha][:l3]["openvswitch_ra"]
+  when ml2_mech_drivers.include?("linuxbridge")
+    neutron_agent = node[:neutron][:platform][:lb_agent_name]
+    neutron_agent_ra = node[:neutron][:ha][:l3]["linuxbridge_ra"]
+  end
 when "vmware"
   neutron_agent = ''
+  neutron_agent_ra = node[:neutron][:ha][:l3]["vmware_ra"]
```

Comment:
Done in https://github.com/crowbar/barclamp-neutron/commit/1504fd042f9cf6bf1e7665d354e3e84e141adf39